model/jpa/src/main/java/org/keycloak/organization/jpa/JpaOrganizationProvider.java

```diff
@@ -108,10 +162,39 @@ private OrganizationAdapter getAdapter(RealmModel realm, String id) {
         }
 
         if (!realm.getId().equals(entity.getRealmId())) {
-            return null;
+            throw new IllegalArgumentException("Organization [" + entity.getId() + " does not belong to realm [" + realm.getId() + "]");
+        }
+
+        return new OrganizationAdapter(entity, session);
+    }
+
+    private GroupModel createOrganizationGroup(RealmModel realm, String name) {
+        String groupName = getCanonicalGroupName(name);
+        GroupModel group = groupProvider.getGroupByName(realm, null, name);
+
+        if (group != null) {
+            throw new IllegalArgumentException("A group with the same already exist and it is bound to different organization");
+        }
+
+        return groupProvider.createGroup(realm, KeycloakModelUtils.generateId(), groupName);
+    }
+
+    private String getCanonicalGroupName(String name) {
+        return "kc.org." + name;
```

Comment:
This could be replaced by
```suggestion
        return OrganizationModel.USER_ORGANIZATION_ATTRIBUTE + name;
```